### PR TITLE
Add auth config to oauth api 

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
New OAuth flows will need values from the connector config in https://github.com/airbytehq/airbyte/issues/6971

## How
Alteernative to https://github.com/airbytehq/airbyte/pull/7678

This API changes adds an "optional" auth config parameter to the API endpoint to be potentially used during oauth flows (depends on flow implementation)

## Context

- When FE calls the completeOauth API endpoint, a auth config is built with values extracted from a partial connector config if any fields is used as input fields to the oauth flow
- once oauth flow is completed, the backend returns the oauth outputs inside the auth config. The Front-end would have to merge this config back to the connector config where it needs to (as declared in the rootObject in auth spec object).
- Once merged, FE can continue with the modified connectionConfig to SourceCreate when clicking on "setup connection"

### CODE REVIEW

**Review by sherifnada — COMMENTED** [review]
I like this approach a lot. It's very well-scoped to the inputs required for oauth. I'd go for this one over #7678

**Review by jamakase — COMMENTED** [review]
Have some questions:

1. How will it handle `airbyte_secret` fields? 
2. I believe we don't yet have file upload, but what if we have it at some point, won't it break this endpoint? Though if we upload file initially and then just send ids - it should be fine. But we need to consider such a case in advance.

I understand that this prop is just additional and we are free to send everything we want here, but the problem is that we can't send exact `connectorConfiguration` property as it may be not valid yet, so there is a possibility that we will send some temp frontend state.

File: airbyte-api/src/main/openapi/config.yaml [hunk 1]
```diff
@@ -3227,6 +3227,8 @@ components:
           items:
             $ref: "#/components/schemas/DbMigrationRead"
     # OAuth
+    AuthInputConfiguration:
```
Comment (michel-tricot) [inline airbyte-api/src/main/openapi/config.yaml:3230]: since it is for OAuth, should it be called OAuth?

**Comment by ChristopheDuong** [comment]
> Have some questions:
> I understand that this prop is just additional and we are free to send everything we want here, but the problem is that we can't send exact `connectorConfiguration` property as it may be not valid yet, so there is a possibility that we will send some temp frontend state.

In this PR, we changed our minds and instead of sending `connectorConfiguration`, we'll be restricting to build a new config object `authInputConfiguration` with only fields that are needed by the OAuth flow as described in its `Input Specification` node.

**Review by sherifnada — APPROVED** [review]

**Review by cgardens — APPROVED** [review]
Awesome! Thanks for the short PR with just the iface change!

File: airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml [hunk 2]
```diff
@@ -305,3 +306,116 @@ definitions:
...
+          Additional and optional specification object to describe what an 'advanced' Auth flow would need to function.
+            - A connector should be able to fully function with the configuration as described by the ConnectorSpecification in a 'basic' mode.
+            - The 'advanced' mode provides easier UX for the user with UI improvements and automations. However, this requires further setup on the
+            server side by instance or workspace admins beforehand. The trade-off is that the user does not have to provide as many technical
+            inputs anymore and the auth process is faster and easier to complete.
+        type: object
+        properties:
+          auth_flow_type:
+            type: string
+            enum: ["oauth1.0", "oauth2.0"] # Future auth types should be added here
```
Comment (ChristopheDuong) [inline airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml:320]:
```suggestion
            enum: ["oauth2.0", "oauth1.0"] # Future auth types should be added here
```

File: airbyte-api/src/main/openapi/config.yaml [hunk 3]
```diff
@@ -3227,6 +3231,115 @@ components:
...
+    OAuthInputConfiguration:
+      description: The values required to configure OAuth flows.
+        The schema for this must match the `OAuthConfigSpecification.oauthUserInputFromConnectorConfigSpecification` schema.
+      $ref: "#/components/schemas/OAuthConfiguration"
+    AdvancedAuth:
+      type: object
+      properties:
+        authFlowType:
+          type: string
+          enum: ["oauth1.0", "oauth2.0"]
```
Comment (ChristopheDuong) [inline airbyte-api/src/main/openapi/config.yaml:3245]:
```suggestion
          enum: ["oauth2.0", "oauth1.0"]
```

**Pull request merged**